internal/pkg/warewulfd/api/container.go

```diff
@@ -0,0 +1,197 @@
+}
+
+func getContainerByName() usecase.Interactor {
+	type getContainerByNameInput struct {
+		Name string `path:"name"`
+	}
+
+	u := usecase.NewInteractor(func(ctx context.Context, input getContainerByNameInput, output *Container) error {
+		if !container.ValidSource(input.Name) {
+			return fmt.Errorf("container not found: %v", input.Name)
+		} else {
+			*output = *NewContainer(input.Name)
+			return nil
+		}
+	})
+	u.SetTitle("Get a container")
+	u.SetDescription("Get a container by its name.")
+	u.SetTags("Container")
+	return u
+}
+
+func importContainer() usecase.Interactor {
+	type importContainerInput struct {
+		Name        string `path:"name"`
+		URI         string `json:"uri" required:"true" description:"Docker registry URI to download container images"`
+		NoHttps     bool   `json:"nohttps" default:"false" description:"Whether to use https for the registry URI, default:'false'"`
+		OciUserName string `json:"ociuser" description:"Username for the registry URI, if needed"`
+		OciPassword string `json:"ocipassword" description:"Password for the registry URI, if needed"`
+		Build       bool   `json:"build" default:"true" description:"Whether to build the container image, default:'true'"`
+		SycUser     bool   `json:"syncuser" default:"true" description:"Whether to syncuser, default:'true'"`
+		Default     bool   `json:"default" default:"false" description:"Whether to make the contaienr default, default:'false'"`
+		Force       bool   `json:"force" default:"false" description:"Whether to overwrite the existing container with the same name, default:'false'"`
+	}
+
+	u := usecase.NewInteractor(func(ctx context.Context, input importContainerInput, output *Container) error {
+		if !strings.HasPrefix(input.URI, "docker://") {
+			return errors.New("uri only supports docker:// prefix for now")
+		}
+
+		cip := &wwapiv1.ContainerImportParameter{
+			Source:      input.URI,
+			Name:        input.Name,
+			Force:       input.Force,
+			Build:       input.Build,
+			Default:     input.Default,
+			OciNoHttps:  input.NoHttps,
+			OciUsername: input.OciUserName,
+			OciPassword: input.OciPassword,
+			SyncUser:    input.SycUser,
+		}
+
+		containerName, err := container_api.ContainerImport(cip)
+		if err != nil {
+			return err
+		}
+
+		*output = *NewContainer(containerName)
+		return nil
+	})
+	u.SetTitle("Import a container")
+	u.SetDescription("Import a container from Docker registry URI")
+	u.SetTags("Container")
+
+	return u
+}
+
+func deleteContainer() usecase.Interactor {
+	type deleteContainerInput struct {
+		Name string `path:"name"`
+	}
+
+	u := usecase.NewInteractor(func(ctx context.Context, input deleteContainerInput, output *Container) error {
+		if !container.ValidSource(input.Name) {
+			return fmt.Errorf("container not found: %v", input.Name)
+		}
+
+		*output = *NewContainer(input.Name)
+		cdp := &wwapiv1.ContainerDeleteParameter{
+			ContainerNames: []string{input.Name},
+		}
+
+		err := container_api.ContainerDelete(cdp)
```

Comment:
Please avoid using the grpc API code inside of this new API code, since the goal is to remove the grpc code. Refactor this (and potentially `container_api.ContainerDelete`) to call a (potentially new) function in `internal/pkg/container/`.